vllm/model_executor/layers/rejection_sampler.py

```diff
@@ -0,0 +1,388 @@
+from typing import Tuple, Optional
+from functools import cached_property
+
+import torch
+import torch.nn as nn
+import torch.jit
+
+
+class RejectionSampler(nn.Module):
+    """Apply modified rejection sampling as described in "Accelerating Large
+        Language Model Decoding with Speculative Sampling"
+        https://arxiv.org/pdf/2302.01318.pdf.
+    """
+
+    def __init__(self, strict_mode: bool = False):
+        """Create a rejection sampler.
+
+        Args:
+            strict_mode: Whether or not to perform shape/device/dtype checks
+                during sampling. This catches correctness issues but adds
+                nontrivial latency.
+        """
+        super().__init__()
+        self.probs_dtype = torch.float32
+        self.token_id_dtype = torch.int64
+        self._num_bonus_tokens = 1
```

Comment:
when can `num_bonus_tokens` > 1? Is it the last generated token by the target model iff all drafted tokens are accepted?

Reply:
> when can num_bonus_tokens > 1?

It is always 1. This variable is for readability only. I'll add a comment.

> Is it the last generated token by the target model iff all drafted tokens are accepted?

Yep!